src/pages/add-product/components/variants-list/index.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import { PenIcon } from "~/assets/icons/pen-icon.tsx";
+import { TrashIcon } from "~/assets/icons/trash-icon.tsx";
+import { VariantItem } from "~/pages/add-product/types.ts";
+
+import { ItemColumn, ItemRow } from "./styles.ts";
+
+type VariantProps = {
+	deleteItem: (id: number) => void;
+	items: VariantItem[];
+	updateItem: (item: VariantItem) => void;
+};
+
+const Variants: React.FC<VariantProps> = ({
+	deleteItem,
+	items,
+	updateItem,
+}) => {
+	const { t } = useTranslation();
+	let itemCount = 1;
```

Comment:
not good solution, use useRef better if you need calculation without rerender, or useState, and it its just index of the element, you can use 2d parameter items.map((item, index)